src/OpenTelemetry.Exporter.Geneva/Metrics/GenevaMetricExporter.cs

```diff
@@ -500,6 +880,99 @@ private void SerializeHistogramBucket(in HistogramBucket bucket, ref int bufferI
         MetricSerializer.SerializeUInt32(this.bufferForHistogramMetrics, ref bufferIndex, Convert.ToUInt32(bucket.BucketCount));
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private void SerializeHistogramBucketWithTLV(in HistogramBucket bucket, ref int bufferIndex, double lastExplicitBound)
+    {
+        if (bucket.ExplicitBound != double.PositiveInfinity)
+        {
+            MetricSerializer.SerializeUInt64(this.buffer, ref bufferIndex, Convert.ToUInt64(bucket.ExplicitBound));
+        }
+        else
+        {
+            // The bucket to catch the overflows is one greater than the last bound provided
+            MetricSerializer.SerializeUInt64(this.buffer, ref bufferIndex, Convert.ToUInt64(lastExplicitBound + 1));
+        }
+
+        MetricSerializer.SerializeUInt32(this.buffer, ref bufferIndex, Convert.ToUInt32(bucket.BucketCount));
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private void SerializeExemplar(Exemplar exemplar, ref int bufferIndex)
+    {
+        MetricSerializer.SerializeByte(this.buffer, ref bufferIndex, 0); // version
+
+        var bufferIndexForLength = bufferIndex;
+        bufferIndex++;
+
+        var bufferIndexForFlags = bufferIndex;
+        bufferIndex++;
+
+        var flags = ExemplarFlags.IsTimestampAvailable; // we only serialize exemplars with Timestamp != default
+
+        // TODO: Update the code whenn Exemplars support long values
+        var value = exemplar.DoubleValue;
+
+        // Check if the double value is actually a whole number that can be serialized as a long instead
+        var valueAsLong = (long)value;
+        bool isWholeNumber = valueAsLong == value;
+        if (isWholeNumber)
+        {
+            flags |= ExemplarFlags.IsMetricValueDoubleStoredAsLong;
+            MetricSerializer.SerializeInt64AsBase128(this.buffer, ref bufferIndex, valueAsLong); // serialize long value
+        }
+        else
+        {
+            MetricSerializer.SerializeFloat64(this.buffer, ref bufferIndex, value); // serialize double value
+        }
+
+        var bufferIndexForNumberOfLabels = bufferIndex;
+        MetricSerializer.SerializeByte(this.buffer, ref bufferIndex, 0); // serialize zero as the count of labels; this would be updated later if the exemplar has labels
+        bool hasLabels = exemplar.FilteredTags != null && exemplar.FilteredTags.Count > 0;
```

Comment:
nit; move hasLabels to further below, closer to the actual place where it is used.